PlaceholderExpansion.md

```diff
@@ -71,265 +74,298 @@ Let's quickly break down the different methods you have to implement.
+### Register your expansion
 
-Your expansion will need to override the `getRequiredPlugin()` method to return the name of the plugin your expansion depends on.  
-PlaceholderAPI automatically checks if this method will either return null, or if the name defined results in a non-null plugin.
+The main downside of an internal expansion is, that it can't be loaded automatically by PlaceholderAPI and instead requires you to manually register it.  
+To do that, create a new instance of your Expansion class and call the `register()` method of it.
 
-It is worth noting that it is a bit more difficult to make a separate jar file that depends on a plugin, as it will require the plugin to have some sort of accessible API in order to get the required values.
-One way to bypass this is to override the `canRegister()` method with the following code:
+Below is an example of loading the Expansion in the Plugin's main class inside the `onEnable()` method.
 
 ```java
-private SomePlugin plugin = null; // This would be the plugin your expansion depends on
+package at.helpch.placeholderapi.example;
 
-@Override
-public boolean canregister() {
-    // This sets plugin to the SomePlugin instance you get through the PluginManager
-    return (plugin = (SomePlugin) Bukkit.getPluginManager().getPlugin(getRequiredPlugin())) != null;
+import at.helpch.placeholderapi.example.expansion.SomeExpansion;
+import org.bukkit.Bukkit;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public class SomePlugin extends JavaPlugin {
+    
+    @Override
+    public void onEnable() {
+        // Make sure PlaceholderAPI is installed and enabled
+        if (Bukkit.getPluginManager().isPluginEnabled("PlaceholderAPI")) {
+            // Register the expansion
+            new SomeExpansion(this).register();
+        }
+    }
 }
 ```
-Using this code-snippet, you can get a direct instance of the plugin and access things such as config values.  
-With that said, it is recommended instead to use an API if one is available, as this kind of plugin access is a relatively poor approach.
 
-#### Full Example
-Please see the [Common parts](#common-parts) section for info on the other methods.
+----
+
+## Making an external Expansion
+
+External expansions are separate Jar files containing the PlaceholderExpansion extending class.  
+They are recommended for the following scenarios:
+
+- Your Expansion doesn't depend on any Plugin.
+- Your Expansion depends on a Plugin AND you can't directly implement the Class into it.
+
+For Expansions made for a plugin is it recommended to try and use [internal Expansions](#making-an-internal-expansion). If you can't implement it yourself, i.e. because you're not the dev of it and it isn't open source, should you consider alternative solutions (i.e. asking the dev to implement it) before using this method.
```

Comment:
May I suggest adding something even more obvious than this to below the header line?

```
> ⚠️ If you'd like to provide PlaceholderAPI expansions for your own plugin, this is not recommended. Instead, [register an internal expansion provider](#making-an-internal-expansion).
```

Reply:
I feel like this would be too biased here.

It already states the main cases this type of expansion can be used for, so I think that's fine enough.

Reply:
It does. Just need a space between the HTML tag and markdown

Reply:
Yeah it should really emphasize that 99.99% of the time, you want to make an internal expansion. Also would be good to add the benefit that you don't need to match versions between the expansion and the plugin, and that setup is easier since it doesn't require downloading from the ecloud.